### ISSUE TITLE
TemporaryFileReaper uses "None"

### ISSUE THREAD

**Issue by mkurz** [playframework/playframework#7569]
I have configured the `TemporaryFileReaper` like this (using just `5 seconds` for testing purposes):
```
play.temporaryFile {
	reaper {
		enabled = true
		initialDelay = "5 seconds"
		interval = "5 seconds"
		olderThan = "30 minutes"
	}
}
```
and set following logger conf:
```
<logger name="play.api.libs.Files" level="DEBUG" />
```

I am getting following output:
```
// starting the application:
[info] p.a.l.Files$DefaultTemporaryFileReaper - Reaper enabled on None, starting in 5 seconds with 5 seconds intervals

// after application started the reaper runs:
[debug] p.a.l.Files$DefaultTemporaryFileReaper - reap: reaping old files from None
[debug] p.a.l.Files$DefaultTemporaryFileReaper - reap: reaping old files from None
[debug] p.a.l.Files$DefaultTemporaryFileReaper - reap: reaping old files from None

// shutting down the application:
[debug] p.a.l.Files$DefaultTemporaryFileCreator - deletePath: deleting = /tmp/playtemp236077686293408167
[debug] p.a.l.Files$DefaultTemporaryFileReaper - reap: reaping old files from Some(/tmp/playtemp236077686293408167)
```

Why does it operate on `None` and not a real temp directory? (`Reaper enabled on `**`None`** and `reaping old files from `**`None`**) When shutting down the reaper suddenly knows about an actual temp folder...

Plus: `deletePath...` runs before the reaper reaps old files from within this directory on shutdown - does this makes sense at all?

I get this log output when running in dev mode as well when staging the application.
FYI: I am running the app without global state (`play.allowGlobalApplication = false`)

**Comment by wsargent** [comment]
I'm guessing that there is a race condition with temp directory existence here:

https://github.com/playframework/playframework/blob/master/framework/src/play/src/main/scala/play/api/libs/Files.scala#L165

**Comment by richdougherty** [comment]
> Why does it operate on None and not a real temp directory? (Reaper enabled onNone and reaping old files fromNone) When shutting down the reaper suddenly knows about an actual temp folder...

I had a look at the code and I noticed that `DefaultTemporaryFileReaper` starts with `private var playTempFolder: Option[Path] = None`. The `playTempFolder` is only updated when there's a call to `DefaultTemporaryFileCreator.create`. The first time `DefaultTemporaryFileCreator.create` is called it sets the reaper's directory. I think this means that the reaper won't have a directory set until a file is created and it will just do nothing. I think that's probably OK. The log messages are only there if you enable debug.

> Plus: deletePath... runs before the reaper reaps old files from within this directory on shutdown - does this makes sense at all?

It looks like `deletePath` is called directly by the `DefaultTemporaryFileCreator`'s application shutdown hook. I'm not sure why the reaper runs immediately after. Perhaps it's a quirk of how the reaping scheduler happens? I don't think it's a problem anyway.

**Comment by richdougherty** [comment]
@wsargent I have created an issue to track the race condition you found: https://github.com/playframework/playframework/issues/7584

**Comment by mkurz** [comment]
> The log messages are only there if you enable debug.

The first log message
```
[info] p.a.l.Files$DefaultTemporaryFileReaper - Reaper enabled on None, starting in 5 seconds with 5 seconds intervals
```
is an info message and you will see it by default. Could maybe confuse people (it confused me - I thought there is a bug).

**Comment by richdougherty** [comment]
OK, perhaps we should make this a `debug` level message, or improve the message so it says no temporary folder has been created yet.

**Comment by richdougherty** [comment]
@mkurz, I believe this should be fixed now in Play 2.6.3. Thanks a lot for all your contributions. You did get a mention in the Credits and the end of the release notes. :) https://blog.playframework.com/play-2-6-3-released/
